pkg/operator/condition/condition.go

```diff
@@ -0,0 +1,56 @@
+
+	// MonitoringResourceControllerDegradedConditionType represents a condition when the operator is unable to create or reconcile the ServiceMonitor
+	// CR resource, which is required by monitoring operator to collect Prometheus data from the operator. When this condition is true and the ServiceMonitor
+	// is already created, it won't have impact on collecting metrics. However, if the ServiceMonitor was not created, the metrics won't be available for
+	// collection until this condition is set to false.
+	MonitoringResourceControllerDegradedConditionType = "MonitoringResourceControllerDegraded"
+
+	// BackingResourceControllerDegradedConditionType represents a condition when the operator is unable to create or reconcile the resources needed
+	// to successfully run the installer pods (installer CRB and SA). If these were already created, this condition is not fatal, however if the resources
+	// were not created it means the installer pod creation will fail.
+	BackingResourceControllerDegradedConditionType = "BackingResourceControllerDegraded"
+
+	// ManagementStateDegradedConditionType represents a condition when the operator observe errors when installing the new revision static pods.
+	// This condition report Error reason when the pods are terminated or not ready or waiting during which the operand quality of service is degraded.
+	StaticPodsDegradedConditionType = "StaticPodsDegraded"
+
+	// ConfigObservationDegradedConditionType represents a condition when the operator failed to observe or process configuration change.
+	// This is not transient condition and normally a correction or manual intervention is required on the config custom resource.
+	ConfigObservationDegradedConditionType = "ConfigObservationDegraded"
+
+	// ResourceSyncControllerDegradedConditionType represents a condition when the operator failed to synchronize one or more secrets or configmaps required
+	// to run the operand. Operand ability to provide service might be affected by this condition.
+	ResourceSyncControllerDegradedConditionType = "ResourceSyncControllerDegraded"
+
+	// CertRotationDegradedConditionType represents a condition when the operator failed to properly rotate one or more certificates required by the operand.
+	// The RotationError reason is given with message describing details of this failure. This condition can be fatal when ignored as the existing certificate(s)
+	// validity can expire and without rotating/renewing them manual recovery might be required to fix the cluster.
+	CertRotationDegradedConditionType = "CertRotation_%s_Degraded"
+
+	// InstallerControllerDegradedConditionType represents a condition when the operator is not able to create new installer pods so the new revisions
+	// cannot be rolled out. This might happen when one or more required secrets or config maps does not exists.
+	InstallerControllerDegradedConditionType = "InstallerControllerDegraded"
```

Comment:
... is not able to create ... is differential again, some event. Please describe also in those cases, when the condition is set back to false again.

Reply:
added the False case to godoc (for others as well).